src/Controller/FilesController.php

```diff
@@ -24,18 +24,21 @@
 use Symfony\Component\HttpFoundation\BinaryFileResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\Security\Core\Security;
 
 class FilesController
 {
     protected $rootDir;
     protected $session;
     protected $framework;
+    private $security;
 
-    public function __construct(string $rootDir, Session $session, ContaoFramework $framework)
+    public function __construct(string $rootDir, Session $session, ContaoFramework $framework, Security $security)
```

Comment:
You are injecting the `Security` service now - however you did not adjust the `services.yml` accordingly. Did you test all of these changes? ;)